core/domain/suggestion_services.py

```diff
@@ -0,0 +1,367 @@
+        suggestion_models.STATUS_IN_REVIEW, author_id, assigned_reviewer_id,
+        final_reviewer_id, change_cmd, score_category, thread_id)
+
+
+def get_suggestion_from_model(suggestion_model):
+    """Converts the given SuggestionModel to a Suggestion object
+
+    Args:
+        suggestion_model: SuggestionModel.
+
+    Returns:
+        Suggestion. The corresponding Suggestion domain object.
+    """
+    return suggestion_domain.Suggestion(
+        suggestion_model.id, suggestion_model.suggestion_type,
+        suggestion_model.target_type, suggestion_model.target_id,
+        suggestion_model.target_version_at_submission,
+        suggestion_model.status, suggestion_model.author_id,
+        suggestion_model.assigned_reviewer_id,
+        suggestion_model.final_reviewer_id, suggestion_model.change_cmd,
+        suggestion_model.score_category)
+
+
+def get_suggestion_by_id(suggestion_id):
+    """Finds a suggestion by the suggestion ID.
+
+    Args:
+        suggestion_id: str. The ID of the suggestion.
+
+    Returns:
+        Suggestion, or None if no suggestion is found.
+    """
+    model = suggestion_models.SuggestionModel.get_by_id(suggestion_id)
+
+    return get_suggestion_from_model(model) if model else None
+
+
+def get_suggestions_by_author(author_id):
+    """Gets a list of suggestions by the given author.
+
+    Args:
+        author_id: str. The author of the suggestions.
+
+    Returns:
+        list(Suggestions): a list of suggestions by the given author.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_author(
+            author_id)]
+
+
+def get_suggestions_reviewed_by(reviewer_id):
+    """Gets a list of suggestions that have been reviewed by the given user.
+
+    Args:
+        reviewer_id: str. The reviewer of the suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions reviewed by the given user.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_reviewed_by(
+            reviewer_id)]
+
+
+def get_suggestions_assigned_to_reviewer(assigned_reviewer_id):
+    """Gets a list of suggestions assigned to the given user for review.
+
+    Args:
+        assigned_reviewer_id: str. The reviewer assigned to review the
+                suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions assigned to the given user
+            for review.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel
+        .get_suggestions_assigned_to_reviewer(assigned_reviewer_id)]
+
+
+def get_suggestions_by_status(status):
+    """Gets a list of suggestions with the given status.
+
+    Args:
+        status: str. The status of the suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions with the given status.
```

Comment:
list(Suggestion). A list ...

(Ditto above/below, and also make sure to format docstrings as typeinfo, followed by a full stop, followed by the description.)

Reply:
Done